### ISSUE TITLE
When using Parallel Route, if the page does not exist in root, the layout of the first slot is not applied.

### ISSUE THREAD

**Issue by lxia1220** [vercel/next.js#52726]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
    Operating System:
      Platform: linux
      Arch: x64
      Version: #22 SMP Tue Jan 10 18:39:00 UTC 2023
    Binaries:
      Node: 16.17.0
      npm: 8.15.0
      Yarn: 1.22.19
      pnpm: 7.1.0
    Relevant Packages:
      next: 13.4.10-canary.8
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 4.9.5
    Next.js Config:
      output: N/A
```

### Link to the code that reproduces this issue or a replay of the bug

https://codesandbox.io/p/sandbox/ecstatic-monad-dh2jyw

### To Reproduce

create 2 level nested parallel routes
only create pages in leaves

- @parentSlot1
- - @childSlot1
- - - layout.tsx
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx       << Render childSlot1, childSlot2
- @parentSlot2
- - @childSlot1
- - - layout.tsx
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx       << Render childSlot1, childSlot2
- layout.tsx << Render parentSlot1, parentSlot2

### Describe the Bug

**Structure**
- @parentSlot1
- - @childSlot1
- - - layout.tsx    -------- Every first slot's layout is not applied
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx 
- @parentSlot2
- - @childSlot1
- - - layout.tsx     -------- Every first slot's layout is not applied
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx 
- layout.tsx

**Render Result**
- layout
- - @parentSlot1/layout
- - - @childSlot1/page
- - - @childSlot2/layout
- - - - @childSlot2/page
- - @parentSlot2/layout
- - - @childSlot1/page
- - - @childSlot2/layout
- - - - @childSlot2/page

### Expected Behavior

- @parentSlot1
- - @childSlot1
- - - layout.tsx    -------- Every first slot's layout should be applied
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx 
- @parentSlot2
- - @childSlot1
- - - layout.tsx     -------- Every first slot's layout should be applied
- - - page.tsx
- - @childSlot2
- - - layout.tsx
- - - page.tsx
- - layout.tsx 
- layout.tsx

**Render Result**
- layout
- - @parentSlot1/layout
- - - @childSlot1/layout
- - - - @childSlot1/page
- - - @childSlot2/layout
- - - - @childSlot2/page
- - @parentSlot2/layout
- - - @childSlot1/layout
- - - - @childSlot1/page
- - - @childSlot2/layout
- - - - @childSlot2/page

**Comment by ctretyak** [comment]
Bumped into the same bug

https://codesandbox.io/p/sandbox/practical-diffie-37nq9l?file=%2Fapp%2F%40xgroup%2Flayout.tsx%3A7%2C2

Layouts in slots just doesn't work correctly. I don't understand their behaviour - sometimes it applies, sometimes doesn't apply

**Comment by patrykszwed** [comment]
Any news on that? Still getting the same behavior on Next `14.0.1`.

A workaround that worked for me was:
1. adding a child slot (that won't be used) named in a way that it's the first one. 
2. adding a `page.tsx` file in that child slot

This is hacky as hell...

**Comment by samcx** [comment]
@lxia1220 Thank you for submitting an issue!

This seems to be <picture data-single-emoji=":fixed:" title=":fixed:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/fixed/66bb8455f24bd50e.png" alt=":fixed:" align="absmiddle"></picture> on the latest <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> canary.

![CleanShot 2024-01-10 at 13 22 34@2x](https://github.com/vercel/next.js/assets/28912696/a7b9b446-750c-436f-9c34-daf004d52573)

Let me know if you can confirm the same!

**Comment by samcx** [comment]
@greghart Great! Thank you for confirming—I'll be closing this issue now!
